bson/src/main/org/bson/codecs/pojo/PojoCodecProvider.java

```diff
@@ -76,14 +80,15 @@ public static Builder builder() {
         } else if (automatic || (clazz.getPackage() != null && packages.contains(clazz.getPackage().getName()))) {
             try {
                 classModel = createClassModel(clazz, conventions);
-            } catch (IllegalStateException e) {
-                return null;
-            }
-            if (!clazz.isInterface() && classModel.getPropertyModels().isEmpty()) {
+                if (!clazz.isInterface() && classModel.getPropertyModels().isEmpty()) {
+                    return null;
+                }
+                discriminatorLookup.addClassModel(classModel);
+                return new AutomaticPojoCodec<T>(new PojoCodecImpl<T>(classModel, registry, propertyCodecProviders, discriminatorLookup));
+            } catch (Exception e) {
+                LOGGER.warn(format("Cannot use '%s' with the PojoCodec. %s", clazz.getSimpleName(), e.getMessage()));
```

Comment:
Generally, exceptions are logged like this so that the exception type and stack trace are also printed:

    LOGGER.warn(format("Cannot use '%s' with the PojoCodec", clazz.getSimpleName()), e);

What do you think about doing that instead?

Reply:
Sounds good - will go with that.